diesel/src/result.rs

```diff
@@ -15,6 +15,11 @@ pub enum Error {
     QueryBuilderError(Box<StdError+Send+Sync>),
     DeserializationError(Box<StdError+Send+Sync>),
     SerializationError(Box<StdError+Send+Sync>),
+    /// You can return this variant inside of a transaction when you want to
+    /// roll it back, but have no actual error to return. Diesel will never
+    /// return this variant unless you gave it to us, and it can be safely
+    /// ignored in error handling.
+    RollbackTransaction,
```

Comment:
Nice docs. While you're at it you could also add a sentence or two to the other variants (even if just to say when this error can occur) :)